charts/replicated-library/templates/lib/_configmap.tpl

```diff
@@ -1,15 +1,28 @@
 {{/*
-Renders the configmap objects required by the chart.
+This template serves as a blueprint for all configMap objects that are created
+within the replicated-library library.
 */}}
-{{- define "replicated-library.configmaps" -}}
-  {{- /* Generate named configmaps as required */ -}}
-  {{- range $name, $configmap := .Values.configmaps }}
-    {{- if $configmap.enabled -}}
-      {{- $configmapValues := $configmap -}}
-
-      {{- $_ := set $ "ObjectValues" (dict "values" $configmapValues) -}}
-
-      {{- include "replicated-library.classes.configmap" $ | nindent 0 }}
-    {{- end }}
+{{- define "replicated-library.configmap" -}}
+  {{- $values := "" -}}
+  {{- if and (hasKey . "ContextValues") (hasKey .ContextValues "configmap") -}}
```

Comment:
If we want to we can simplify each of these if statements. The template sets the "ContextValues" explicitly and I think it's ok that templates take that as a contract that this value always exists. 

That would allow us to compact each of these statements to a more simple check of the key rather than verifying that the variable exists.

However, I do understand this is good practice and if you would prefer to keep it to encourage good template practices I'm fine with that.

Reply:
Yeah, I think I like the fact it does this only because I still want to explore one day making each template independent.